types/index.d.ts

```diff
@@ -0,0 +1,335 @@
+export interface JoystickManagerOptions {
+    /**
+     * Defaults to `'body'`
+     * The dom element in which all your joysticks will be injected.
+     *
+     * This zone also serve as the mouse/touch events handler.
+     *
+     * It represents the zone where all your joysticks will be active.
+     */
+    zone?: HTMLElement;
+
+    /**
+     * Defaults to `'white'`
+     * The background color of your joystick’s elements.
+     *
+     * Can be any valid CSS color.
+     */
+    color?: string;
+
+    /**
+     * Defaults to `100`
+     * The size in pixel of the outer circle.
+     *
+     * The inner circle is 50% of this size.
+     */
+    size?: number;
+
+    /**
+     * This is the strength needed to trigger a directional event.
+     *
+     * Basically, the center is 0 and the outer is 1.
+     *
+     * You need to at least go to 0.1 to trigger a directional event.
+     */
+    threshold?: number;
+
+    /**
+     * Defaults to `250`
+     *
+     * The time it takes for joystick to fade-out and fade-in when activated or de-activated.
+     */
+    fadeTime?: number;
+
+    /**
+     * Defaults to `false`
+     *
+     * Enable the multitouch capabilities.
+     *
+     * If, for reasons, you need to have multiple nipples into the same zone.
+     *
+     * Otherwise it will only get one, and all new touches won’t do a thing.
+     *
+     * Please note that multitouch is off when in static or semi modes.
+     */
+    multitouch?: boolean;
+
+    /**
+     * Defaults to `1`
+     *
+     * If you need to, you can also control the maximum number of instance that could be created.
+     *
+     * Obviously in a multitouch configuration.
+     */
+    maxNumberOfNipples?: number;
+
+    /**
+     * Defaults to `false`
+     *
+     * The library won’t draw anything in the DOM and will only trigger events with data.
+     */
+    dataOnly?: boolean;
+
+    /**
+     * Defaults to `{top: 0, left: 0}`
+     *
+     * An object that will determine the position of a static mode.
+     *
+     * You can pass any of the four top, right, bottom and left.
+     *
+     * They will be applied as any css property.
+     */
+    position?: Position;
```

Comment:
This is different from the x,y Position type defined below. They are css properties, so it should be an object with strings. Need to fix this.